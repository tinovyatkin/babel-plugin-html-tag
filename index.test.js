"use strict";

const { readFileSync } = require("fs");
const path = require("path");

const babel = require("@babel/core");

const plugin = require("./index");

function transform(code) {
  return babel.transform(code, {
    plugins: [plugin],
  }).code;
}

describe("babel-plugin-html-tag", () => {
  it("returns tagged template", () => {
    const res = transform("html`<p>\n${ textToTest }\n</p >`");
    console.log(res);
    expect(res).toMatchSnapshot();
  });

  it("returns static string when possible", () => {
    const res = transform("html`<p class=\"center\">AAA bbb, 'fff'</p>`");
    console.log(res);
    expect(res).toMatchSnapshot();
  });

  it("fixture test", () => {
    const code = readFileSync(
      path.resolve(__dirname, "./__fixtures__/a.js"),
      "utf8"
    );
    expect(transform(code)).toMatchSnapshot();
  });

  it("style tag with generated content", () => {
    const res = transform(`html\`
    <style>
      @media only screen and (min-width: 768px) {
        \${fieldsets
          .map(
            className =>
              \`fieldset.\${className} { background-image: url(\${basePath}\${className}-1400px.\${ext}); }\`,
          )
          .join('\\n')}
      }
    </style>\`
    `);
    expect(res).toMatchSnapshot();
  });
});
