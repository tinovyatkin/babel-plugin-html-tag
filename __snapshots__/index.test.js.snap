// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-html-tag fixture test 1`] = `
"\\"use strict\\";
/* global html */

const b = \\"test\\";
const a = \`<p class=\\"zoom center justify\\">This is paragraph with \${b} subsitutions at several lines: \${1 + 2}</p>\`;
const z = '<table class=\\"center\\"><tr class=\\"left\\"><td>HTML without substitutions</td></tr></table>';
const e = '<style>[data-clipboard-text]{cursor:copy!important;transition:all .5s cubic-bezier(.19,1,.22,1)}[data-clipboard-text]:hover{background-color:green!important;color:#fff!important}</style>';
const n = '<div ng-class=\\"{\\\\'my-class\\\\': myVar}\\"></div>';
const classNames = {};
const t = {
  item(data) {
    return strToEl(\`<div class=\\"\${String(classNames.item)} \${String(data.highlighted ? classNames.highlightedState : classNames.itemSelectable)}\\" data-item data-id=\\"\${String(data.id)}\\" data-value=\\"\${String(data.value)}\\" \${String(data.active ? 'aria-selected=\\"true\\"' : \\"\\")}><span>\${String(data.label)} </span>&nbsp;&nbsp;( <strong class=\\"mono\\" style=\\"font-size:110%\\">\${String(data.value)} </strong>)</div>\`);
  },

  choice(data) {
    return strToEl(\`<div class=\\"nobr \${String(classNames.item)} \${String(classNames.itemChoice)} \${classNames.itemSelectable}\\" data-choice data-choice-selectable data-id=\\"\${String(data.id)}\\" data-value=\\"\${String(data.value)}\\" role=\\"option\\"><span>\${String(data.label)}</span> <strong class=\\"mono\\">\${String(data.value)}</strong></div>\`);
  },

  option(data) {
    return strToEl(\`<option value=\\"\${data.value}\\" selected=\\"selected\\" data-lat=\\"\${data.customProperties.pos[1]}\\" data-lng=\\"\${data.customProperties.pos[0]}\\" data-place-id=\\"\${data.customProperties.placeId}\\">\${data.label}</option>\`);
  }

};
const st = \`<style>@media only screen and (min-width: 768px) {
      \${fieldsets.map(className => \`fieldset.\${className} { background-image: url(\${basePath}\${className}-1400px.\${ext}); }\`).join(\\"\\\\n\\")}
    }</style><style>.byaka {
      background: \${eee};
    }</style>\`;"
`;

exports[`babel-plugin-html-tag returns static string when possible 1`] = `"'<p class=\\"center\\">AAA bbb, \\\\'fff\\\\'</p>';"`;

exports[`babel-plugin-html-tag returns tagged template 1`] = `"\`<p>\${textToTest}</p>\`;"`;

exports[`babel-plugin-html-tag style tag with generated content 1`] = `
"\`<style>@media only screen and (min-width: 768px) {
        \${fieldsets.map(className => \`fieldset.\${className} { background-image: url(\${basePath}\${className}-1400px.\${ext}); }\`).join('\\\\n')}
      }</style>\`;"
`;
